File: ontrack-acceptance/src/test/java/net/nemerosa/ontrack/acceptance/ACCGeneral.java
package net.nemerosa.ontrack.acceptance;

import com.fasterxml.jackson.databind.JsonNode;
import org.apache.commons.lang3.StringUtils;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class ACCGeneral extends AcceptanceTestClient {

    @Test
    public void info_api() {
        anonymous().get("info").withNode(node -> {
            JsonNode version = node.path("version");
            assertTrue(StringUtils.isNotBlank(version.path("full").asText()));
            assertTrue(StringUtils.isNotBlank(version.path("base").asText()));
            assertTrue(StringUtils.isNotBlank(version.path("build").asText()));
            assertTrue(StringUtils.isNotBlank(version.path("commit").asText()));
            assertTrue(StringUtils.isNotBlank(version.path("source").asText()));
        });
    }

}

File: ontrack-kdsl-acceptance/src/test/java/net/nemerosa/ontrack/kdsl/acceptance/tests/core/ACCGeneral.kt
package net.nemerosa.ontrack.kdsl.acceptance.tests.core

import net.nemerosa.ontrack.kdsl.acceptance.tests.AbstractACCDSLTestSupport
import org.junit.jupiter.api.Test
import org.springframework.web.client.HttpClientErrorException
import kotlin.test.assertFailsWith
import kotlin.test.assertTrue

class ACCGeneral : AbstractACCDSLTestSupport() {

    @Test
    fun info_api() {
        val info = ontrack.connector.get("/rest/info").body.asJson()
        assertTrue(
                info.path("version").path("display").asText().isNotBlank(),
                "Version is filled in"
        )
    }

    /**
     * Checks that the client received a HTTP 403 when he tries to access a non-authorized resource.
     */
    @Test
    fun access_control() {
        assertFailsWith<HttpClientErrorException.Unauthorized> {
            ontrack.connector.get("/rest/admin/logs", noAuth = true)
        }
    }

}
